### PR TITLE
frodo playlists support

### PR DESCRIPTION
**Pull request by weekendsource**
Playlist wouldn't play in frodo. This is a quick and dirty fix for "play all" when selecting a playlist in frodo.

### CODE REVIEW

File: resources/lib/kodion/impl/xbmc/xbmc_playlist.py [hunk 1]
```diff
@@ -25,7 +25,13 @@ def clear(self):
 
     def add(self, base_item):
         item = xbmc_items.to_item(self._context, base_item)
-        self._playlist.add(base_item.get_uri(), listitem=item)
+        # frodo fix
+        if self._context.get_system_version().get_version() > (12, 3):
+            self._playlist.add(base_item.get_uri(), listitem=item)
```
Comment (bromix) [inline resources/lib/kodion/impl/xbmc/xbmc_playlist.py]: Redundant code...I think we don't need this ;) line 33 does the same.

File: resources/lib/kodion/impl/xbmc/xbmc_playlist.py [hunk 2]
```diff
@@ -25,7 +25,13 @@ def clear(self):
...
+            pass
+        elif item:
```
Comment (bromix) [inline resources/lib/kodion/impl/xbmc/xbmc_playlist.py]: Why check for ```not None``` but access the value in line 30?

File: resources/lib/kodion/impl/xbmc/xbmc_context_ui.py [hunk 3]
```diff
@@ -22,7 +22,10 @@ def __init__(self, xbmc_addon, context):
 
     def create_progress_dialog(self, heading, text=None, background=False):
         if background:
-            return XbmcProgressDialogBG(heading, text)
+            # frodo fix
+            if self._context.get_system_version().get_version() > (12, 3):
```
Comment (bromix) [inline resources/lib/kodion/impl/xbmc/xbmc_context_ui.py]: I seee, Frodo hasn't a background progress?

I think

```python
if background and self._context.get_system_version().get_version() > (12, 3):
    return XbmcProgressDialogBG(heading, text)
```

looks nicer ;)

Thanks, I will update the framework with this.

File: resources/lib/kodion/impl/xbmc/xbmc_playlist.py (hunk 2 above)
Comment (weekendsource) [inline resources/lib/kodion/impl/xbmc/xbmc_playlist.py]: Yes, it is utter stupid and redundant :))) It should be something like 

    def add(self, base_item):
        item = xbmc_items.to_item(self._context, base_item)
        if item :
          self._playlist.add(base_item.get_uri(), listitem=item)
        pass


 The whole deal was for playlists longer than 50 videos, if any longer frodo spits:

                                              File "C:\Documents and Settings\Weekend\Application Data\XBMC\addons\plugin.video.youtube\resources\lib\youtube\helper\yt_play.py", line 139, in play_playlist
                                                playlist.add(video)
                                              File "C:\Documents and Settings\Weekend\Application Data\XBMC\addons\plugin.video.youtube\resources\lib\kodion\impl\xbmc\xbmc_playlist.py", line 28, in add
                                                self._playlist.add(base_item.get_uri(), listitem=item)
                                            TypeError: Non api type passed to "add" in place of the expected type "p.XBMCAddon::xbmcgui::ListItem."
                                            -->End of Python script error report<--

I even don't know how this abomination got here, I'm new at github so I guess I messed up something with source :)

File: resources/lib/kodion/impl/xbmc/xbmc_context_ui.py (hunk 3 above)
Comment (weekendsource) [inline resources/lib/kodion/impl/xbmc/xbmc_context_ui.py]: It sure does, I said it was quick and dirty :)))  DialogProgressBG is added in gotham. Thank you for your time :)

Comment (bromix) [inline resources/lib/kodion/impl/xbmc/xbmc_context_ui.py]: No problem..I get the picture of you PR...I try to add the change for Frodo to the weekend.

Comment (weekendsource) [inline resources/lib/kodion/impl/xbmc/xbmc_context_ui.py]: Thanks :)

**Pull request merged**